File: frontend/console/src/api/signal-virtual.js
import Axios from "@/http";

export function createSignalVirtual(params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-virtual', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function retrieveSignalVirtual(signalVirtualId) {
    return new Promise((resolve, reject) => {
        Axios.get(`/config/signal-virtual/${ signalVirtualId }`).then(res => {
            resolve(res)
        })
    })
}

export function updateSignalVirtual(params) {
    return new Promise((resolve, reject) => {
        Axios.put('/config/signal-virtual', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalVirtual(signalVirtualId) {
    return new Promise((resolve, reject) => {
        Axios.delete(`/config/signal-virtual/${ signalVirtualId }`).then(res => {
            resolve(res)
        })
    })
}

export function deleteSignalVirtualWithFlag(signalVirtualId) {
    return new Promise((resolve, reject) => {
        Axios.put(`/config/signal-virtual/${ signalVirtualId }`).then(res => {
            resolve(res)
        })
    })
}

export function getAllSignalVirtuals() {
    return new Promise((resolve, reject) => {
        Axios.get('/config/signal-virtual/all').then(res => {
            resolve(res)
        })
    })
}

export function querySignalVirtual (params) {
    return new Promise((resolve, reject) => {
        Axios.post('/config/signal-virtual/query', params, { isJSON: true }).then(res => {
            resolve(res)
        })
    })
}

File: little-bee/controller/router/signal_virtual.go
package router

import (
	"controller/config"

	"github.com/gin-gonic/gin"
)

const URL_ROUTER_SIGNAL_VIRTUAL = "/config/signal-virtual"
const URL_ROUTER_SIGNAL_VIRTUAL_BY_ID = "/config/signal-virtual/:signal-virtual-id"
const URL_ROUTER_SIGNAL_VIRTUAL_QUERY = "/config/signal-virtual/query"
const URL_ROUTER_SIGNAL_VIRTUAL_ALL = "/config/signal-virtual/all"

func SignalVirtualRouter(r *gin.RouterGroup) *gin.RouterGroup {
	r.POST(URL_ROUTER_SIGNAL_VIRTUAL, config.CreateSignalVirtual)
	r.GET(URL_ROUTER_SIGNAL_VIRTUAL_BY_ID, config.RetrieveSignalVirtual)
	r.PUT(URL_ROUTER_SIGNAL_VIRTUAL, config.UpdateSignalVirtual)
	r.DELETE(URL_ROUTER_SIGNAL_VIRTUAL_BY_ID, config.DeleteSignalVirtual)
	r.POST(URL_ROUTER_SIGNAL_VIRTUAL_QUERY, config.QuerySignalVirtual)
	r.GET(URL_ROUTER_SIGNAL_VIRTUAL_ALL, config.GetAllSignalVirtuals)
	return r
}